ic-utils/src/canister.rs

```diff
@@ -192,6 +193,25 @@ impl Argument {
         }
     }
 
+    /// Add an IDLValue Argument. If the current value of Argument is Raw, will set the
+    /// result to an error. If the current value is an error, will do nothing.
+    pub fn push_value_arg(&mut self, arg: IDLValue) {
```

Comment:
`IDLValue` implements CandidType, so you should be able to use `push_idl_arg` with an `IDLValue`. Was there any issue with using that other function?

Reply:
Yes, `IDLValue::ty()` calls unreachable() : https://github.com/dfinity/candid/blob/master/rust/candid/src/parser/value.rs#L326
called from here: https://github.com/dfinity/candid/blob/master/rust/candid/src/ser.rs#L28